common/flatpak-dir.c

```diff
@@ -2458,10 +2459,19 @@ flatpak_dir_mark_changed (FlatpakDir *self,
                           GError    **error)
 {
   g_autoptr(GFile) changed_file = NULL;
+  g_autofree char * changed_path = NULL;
 
   changed_file = flatpak_dir_get_changed_path (self);
+  changed_path = g_file_get_path (changed_file);
```

Comment:
Hmm, good suggestion, but looking at the docs, as g_file_peek_path is a cache, it would only confer a benefit if anything else used it - however this GFile is newly-allocated every time the changed path is accessed.

Reply:
The docs also (correctly) say “It also avoids an extra duplicated string where possible”, which precisely true in the case of a local file.

Reply:
i.e. In the case of a local file, no caching is used, and the internal path is returned directly.

Reply:
These ones don't: https://developer.gnome.org/gio/stable/GFile.html#g-file-peek-path

Anyway: we don't have that function.

Reply:
We should add that function (at some point). It’s pretty chill.